docs/adding-new-resource.md

```diff
@@ -0,0 +1,705 @@
+
+An example would be [`aws_vpc`] and [here][vpc-config] is its configuration.
+
+### Case 4: Random Identifier Substring From Provider
+
+The ID used in the `Import` section is partially random and assigned by the
+provider. For example, a node in a cluster could have a random ID like `13213`
+but the Terraform Identifier could include the name of the cluster that's
+represented as an argument field under `Argument Reference`, i.e.
+`cluster-name:23123`. In that case, we'll use only the randomly assigned part
+as external name and we need to tell Upjet how to construct the full ID back
+and forth.
+
+```golang
+func resourceName() config.ExternalName{
+  e := config.IdentifierFromProvider
+  e.GetIDFn = func(_ context.Context, externalName string, parameters map[string]interface{}, _ map[string]interface{}) (string, error) {
+		cl, ok := parameters["cluster_name"]
+		if !ok {
+			return "", errors.New("cluster_name cannot be empty")
+		}
+		return fmt.Sprintf("%s:%s", cl.(string), externalName), nil
+	}
+	e.GetExternalNameFn = func(tfstate map[string]interface{}) (string, error) {
+		id, ok := tfstate["id"]
+		if !ok {
+			return "", errors.New("id in tfstate cannot be empty")
+		}
+		w := strings.Split(s.(string), ":")
+		return w[len(w)-1], nil
+	}
+}
+```
+
+### Case 5: Non-random Substrings as Identifier
+
+There are more than a single argument under `Argument Reference` that are
+concatenated to make up the whole identifier, e.g. `<region>/<cluster
+name>/<node name>`. We will need to tell Upjet to use `<node name>` as external
+name and take the rest from the parameters.
+
+Use `config.TemplatedStringAsIdentifier("<name argument>", "<go template>")` in
+such cases. The following is the list of available parameters for you to use in
+your go template:
+
+```
+parameters: A tree of parameters that you'd normally see in a Terraform HCL
+            file. You can use TF registry documentation of given resource to
+            see what's available.
+
+terraformProviderConfig: The Terraform configuration object of the provider. You can
+                take a look at the TF registry provider configuration object
+                to see what's available. Not to be confused with ProviderConfig
+                custom resource of the Crossplane provider.
+
+externalName: The value of external name annotation of the custom resource.
+              It is required to use this as part of the template.
+```
+
+You can see example usages in the big three providers below.
+
+#### AWS
+
+For `aws_glue_user_defined_function`, we see that the `name` argument is used
+to name the resource and the import instructions read as the following:
+
+```
+Glue User Defined Functions can be imported using the
+`catalog_id:database_name:function_name`. If you have not set a Catalog ID
+specify the AWS Account ID that the database is in, e.g.,
+
+$ terraform import aws_glue_user_defined_function.func 123456789012:my_database:my_func
+```
```

Comment:
```suggestion
> Glue User Defined Functions can be imported using the
> `catalog_id:database_name:function_name`. If you have not set a Catalog ID
> specify the AWS Account ID that the database is in, e.g.,
>
> `$ terraform import aws_glue_user_defined_function.func 123456789012:my_database:my_func`
```

Suggestion: A block quote seems more appropriate here and has the additional benefit of formatting nested code properly.